src/Compilers/CSharp/Portable/Syntax/NullableDirectiveMap.cs

```diff
@@ -9,26 +9,43 @@
 
 namespace Microsoft.CodeAnalysis.CSharp.Syntax
 {
-    internal sealed class NullableDirectiveMap
+    /// <summary>
+    /// Contains the nullable warnings and annotations context values at a given position in source,
+    /// where <see langword="true"/> means the context is 'enable', <see langword="false"/> means the context is 'disable',
+    /// and <see langword="null"/> means the context is 'restore' or not specified.
+    /// </summary>
+    internal readonly struct NullableDirective
```

Comment:
I would call this something else, because it doesn't really represent a NullableDirective. Maybe NullableContextState?

Reply:
OK, using `PragmaWarningStateMap` as a precedent, how about we change the name `NullableDirective` to `NullableContextState`, and change `NullableDirectiveMap` to `NullableContextStateMap`.